api/v1alpha4/awsmachine_types.go

```diff
@@ -147,6 +147,12 @@ type AWSMachineSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum:=default;dedicated;host
 	Tenancy string `json:"tenancy,omitempty"`
+
+	// AMIType defines the AMI type. AmazonLinuxGPU type is only available for EKS
+	// +kubebuilder:validation:Enum:=AmazonLinux;AmazonLinuxGPU
+	// +kubebuilder:default:=AmazonLinux
+	// +optional
+	AMIType *MachineAMIType `json:"amiType,omitempty"`
```

Comment:
Not sure about adding `AMIType` when we already have an `AMI` field. It makes for a confusing UX.

I would rather change `AMI` from `AWSResourceReference` to a new `AMIReference` and add the EKS option there. Note that since we support Ubuntu, Amazon Linux, CentOS, and we will support Flatcar soon, having a field that only allows either AmazonLinux or AmazonLinuxGPU is also confusing.

If we have the customise AMIReference type, we get something like:

```
// AMIReference is a reference to a specific AWS resource by ID, ARN, or filters.
// Only one of ID, ARN or Filters may be specified. Specifying more than one will result in
// a validation error.
type AMIReference struct {
	// ID of resource
	// +optional
	ID *string `json:"id,omitempty"`

	// ARN of resource
	// +optional
	ARN *string `json:"arn,omitempty"`

	// Filters is a set of key/value pairs used to identify a resource
	// They are applied according to the rules defined by the AWS API:
	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Filtering.html
	// +optional
	Filters []Filter `json:"filters,omitempty"`

        // EKSOptimizedLookupType If specified, will look up an EKS Optimized image in SSM Parameter store
        // +kubebuilder:validation:Enum:=AmazonLinux;AmazonLinuxGPU
        EKSOptimizedLookupType *EKSAMILookupType `json:"eksLookupType,omitempty"`
}
```

Note, there should be no default here, as we default to not using the optimised image type via nil.

Reply:
I like this idea of `AMIReference` as its more explicit and easier to understand. So a 

+1

Reply:
`EKSOptimizedLookupType` field should be optional, right? If user is not providing any `EKSOptimizedLookupType` then it should use AmazonLinux optimised Non GPU AMIs from path `/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended/image_id` exactly like the current behaviour

Reply:
For EKS, yes it should.

Reply:
Good catch, thanks!